MerlinAU.sh

```diff
@@ -5293,25 +5292,44 @@ _EntwareServicesHandler_()
 }
 
 ##------------------------------------------##
-## Modified by ExtremeFiretop [2024-Oct-13] ##
+## Modified by ExtremeFiretop [2024-Nov-15] ##
 ##------------------------------------------##
 _GetOfflineFirmwareVersion_()
 {
     local zip_file="$1"
     local extract_version_regex='[0-9]+_[0-9]+\.[0-9]+_[0-9a-zA-Z]+'
     local validate_version_regex='[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+(_[0-9a-zA-Z]+)?'
     local fwVersionFormat
+    local firmware_version
+    local formatted_version
 
     # Extract the version number using regex #
     firmware_version="$(echo "$zip_file" | grep -oE "$extract_version_regex")"
 
-    if [ -n "$firmware_version" ]
-    then
-        # Replace underscores with dots and insert .0 before the suffix #
-        formatted_version="$(echo "$firmware_version" | sed -E 's/^([0-9]+)_([0-9]+\.[0-9]+)_([0-9a-zA-Z]+)/\1.\2.0_\3/')"
+    if [ -n "$firmware_version" ]; then
+        if echo "$firmware_version" | grep -qE '^([0-9]+)_([0-9]+)\.([0-9]+)_([0-9]+)$'; then
+            # Numeric patch version
+            formatted_version="$(echo "$firmware_version" | sed -E 's/^([0-9]+)_([0-9]+)\.([0-9]+)_([0-9]+)/\1.\2.\3.\4/')"
+        elif echo "$firmware_version" | grep -qE '^([0-9]+)_([0-9]+)\.([0-9]+)_([0-9a-zA-Z]+)$'; then
+            # Alphanumeric suffix
+            formatted_version="$(echo "$firmware_version" | sed -E 's/^([0-9]+)_([0-9]+)\.([0-9]+)_([0-9a-zA-Z]+)/\1.\2.\3.0_\4/')"
+        else
+            printf "\nFailed to parse firmware version from the ZIP file name.\n"
+            firmware_version=""
+        fi
```

Comment:
Additional regex modifications for offline checking... I know I just touched this on PR: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/342

However there is still an issue where versions such as the latest beta 3004.388.8_3 where it's not correctly tagged:

![image](https://github.com/user-attachments/assets/53748819-f571-4c06-a535-fb0f6515aa38)

Reply:
Confirmed on this latest alpha it works much better for me! :)
**Before:**
![image](https://github.com/user-attachments/assets/21bdf923-8c7b-46db-af67-6f6c3bc7676a)

**After:**
![image](https://github.com/user-attachments/assets/f7b7af54-a261-4cc9-a079-928758f53fd6)